src/databricks/labs/lsql/dashboards.py

```diff
@@ -144,6 +149,170 @@ def from_path(cls, path: Path) -> "WidgetMetadata":
         return fallback_metadata.replace_from_arguments(shlex.split(first_comment))
 
 
+class Tile:
+    """A dashboard tile."""
+
+    def __init__(
+        self,
+        name: str,
+        content: str,
+        *,
+        position: Position = Position(0, 0, 0, 0),
+    ) -> None:
+        self._name = name
+        self._content = content
+        self.position = position
+
+        width, height = self._default_size
+        self.position = dataclasses.replace(position, width=position.width or width, height=position.height or height)
+
+    @property
+    def _default_size(self) -> tuple[int, int]:
+        return 0, 0
+
+    def place_after(self, position: Position) -> "Tile":
+        """Place the tile after another tile:
+
+        The tiling logic works if:
+        - `position.width < _MAXIMUM_DASHBOARD_WIDTH` : tiles in a single row should have the same size
+        - `position.width == _MAXIMUM_DASHBOARD_WIDTH` : any height
+        """
+        x = position.x + position.width
+        if x + self.position.width > _MAXIMUM_DASHBOARD_WIDTH:
+            x = 0
+            y = position.y + position.height
+        else:
+            y = position.y
+        new_position = dataclasses.replace(self.position, x=x, y=y)
+
+        replica = copy.deepcopy(self)
+        replica.position = new_position
+        return replica
+
+    @property
+    def widget(self) -> Widget:
+        widget = Widget(name=self._name, textbox_spec=self._content)
+        return widget
+
+
+class MarkdownTile(Tile):
+    @property
+    def _default_size(self) -> tuple[int, int]:
+        return _MAXIMUM_DASHBOARD_WIDTH, 2
+
+
+class QueryTile(Tile):
+    @staticmethod
```

Comment:
Split:
1. Private method that returns the query abstract syntax
2. Use that in the find fields